en/material-system/Material-upgrade-documentation-for-v3.0-to-v3.1.md

```diff
@@ -0,0 +1,157 @@
+        
+            // Fill in your data here
+
+            gl_Position = fill in your data result;
+        }
+    }%
+    ```
+
+### 1.3 Slice shaders
+
+- `CC_STANDARD_SURFACE_ENTRY()`
+
+    Load the standard shader header file `standard-surface-entry` and use the v3.1 standard shader output function `CC_STANDARD_SURFACE_ENTRY()` to replace the original v3.0 shader output function `frag()`.
+
+    ```c
+    CCProgram standard-fs %{
+  
+        // Include your headfile
+   
+        #include <cc-fog-fs> // Note the change in the header file name here
+        #include <standard-surface-entry> // Note the change in the name of the standard shader header file here
+
+        // Fill in your data here
+
+        void surf (out StandardSurface s) {
+ 
+            // Fill in your data here
+
+        }
+        CC_STANDARD_SURFACE_ENTRY() // Standard shader output function
+    }%
+    ```
+
+## 2. Deferred Rendering Pipeline
+
+### 2.1 Deferred Rendering Pipeline
+
+The biggest difference between v3.1 and v3.0 is that v3.1 supports the deferred render pipeline. The engine comes with a standard `standard-surface-entry` header file that supports both the forward render pipeline and the deferred render pipeline, which is used as follows:
+
+```c
+CCEffect %{
+    techniques:
+
+    // Fill in your data here
+
+        - &deferred
+        vert: // your Vertex shader
+        frag: // your Fragment shader
+        phase: deferred
+        propertyIndex: 0
+        blendState:
+            targets: // turn off blending
+            - blend: false
+            - blend: false
+            - blend: false
+            - blend: false
+            properties: // your properties name
+
+    // Fill in your data here
+            
+}%
+
+// fill in your data here
+
+CCProgram standard-fs %{
+    precision highp float;
+    #include <cc-global>
+    #include <shared-ubos>
+    #include <cc-fog-fs> // Note the change in the header file name here.
+    #include <standard-surface-entry> // Note the change in the name of the standard shader header file here
+
+    // Fill in your data here
+    void surf (out StandardSurface s) {
+
+        // Fill in your data here
+
+    }
+    CC_STANDARD_SURFACE_ENTRY() // Standard shader output function
+}%
+
+// fill in your data here
+
+```
+
+### 2.2 Render pipeline determination
+
+The header file `standard-surface-entry` determines which render pipeline is selected, and the lighting calculation is in the file `shading-standard-additive`.
+
+If it is a deferred render pipeline, the `deferred-lighting` effect file is called first, followed by the light calculation file `shading-standard-additive`.
+
+```c
+#define CC_STANDARD_SURFACE_ENTRY()                                 
+#if CC_FORWARD_ADD                                                 
+  #include <shading-standard-additive>
+
+  // Fill in your data here
+
+#elif CC_PIPELINE_TYPE == CC_PIPELINE_TYPE_FORWARD  // Determine if it is the forward render pipeline
```

Comment:
this doesn't render correctly

Reply:
![image](https://user-images.githubusercontent.com/55615178/120758289-3904dc80-c544-11eb-9ad1-837c366b2fe9.png)
What do you mean?

Reply:
oh, wow. I apologize. It didn't render correctly when I was reviewing the language.